include/dlaf/permutations/general/impl.h

```diff
@@ -126,9 +129,51 @@ void applyPermutations(GlobalElementIndex out_begin, GlobalElementSize sz, SizeT
   }
 }
 
+template <class T, Device D, Coord C>
+void applyPermutationsFiltered(
+    GlobalElementIndex out_begin, GlobalElementSize sz, SizeType in_offset,
+    const matrix::Distribution& subm_dist, const SizeType* perm_arr,
+    const std::vector<pika::shared_future<matrix::Tile<const T, D>>>& in_tiles_fut,
+    const std::vector<matrix::Tile<T, D>>& out_tiles, std::function<bool(SizeType)> filter) {
```

Comment:
Regarding the duplication here, you could do:
```suggestion
    const std::vector<matrix::Tile<T, D>>& out_tiles, std::function<bool(SizeType)> filter = [](SizeType){ return false; }) {
```
If you also template the callable the compiler might completely optimize it away. The downside is another template parameter, and if you want to check that the callable is a `bool(SizeType)` the signature becomes more verbose.